index.js

```diff
@@ -106,35 +106,41 @@ exports.generate = async (inputFile, outputDirectory, isApiMonolith, userProvide
 				TYPES_DIRECTORY: isApiMonolith ? "../apiModelTypes" : "./apiModelTypes",
 				BASE_PATH: serviceFileBasePath,
 				FUNCTIONS: _.map(paths, (pathConfig) => {
+					const pathParams = _.filter(pathConfig.parameters, (parameter) => {
+						return _.toLower(parameter.in) === "path";
+					});
 					const queryParams = _.filter(pathConfig.parameters, (parameter) => {
-						return parameter.in === "query";
+						return _.toLower(parameter.in) === "query";
 					});
 
-					const headQueryName = !_.isEmpty(queryParams) && _.head(queryParams).name;
-
 					return {
 						FUNCTION_SUMMARY: pathConfig.summary,
 						FUNCTION_NAME:
 							pathConfig.operationId ||
 							generateOperationId(pathConfig.method, pathConfig.path),
 						FUNCTION_PARAMS: pathConfig.parameters && {
-							FUNCTION_PARAM_CONFIGS: _.map(pathConfig.parameters, (parameter) => {
+							FUNCTION_PARAM_CONFIGS: _.map(pathParams, (parameter) => {
 								return {
 									FUNCTION_PARAM: parameter.name,
-									FUNCTION_PARAM_DESCRIPTION: parameter.description || "stub"
+									FUNCTION_PARAM_DESCRIPTION: parameter.description || "stub",
+									FUNCTION_PARAM_OPTIONAL: parameter.required
+								};
+							})
+						},
+						QUERY_PARAMS: queryParams && {
+							QUERY_PARAM_CONFIGS: _.map(queryParams, (parameter) => {
+								return {
+									QUERY_PARAM: parameter.name,
+									QUERY_PARAM_DESCRIPTION: parameter.description || "stub",
+									QUERY_PARAM_OPTIONAL: parameter.required
```

Comment:
- [ ] This value sounds like it will produce the opposite result

Reply:
Fixed